types/chain_info.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// ChainInfo is used to track metadata about a peer and its chain.
+type ChainInfo struct {
```

Comment:
Now we are on the same page!! In #3167 I have defined a similar type [here](https://github.com/filecoin-project/go-filecoin/pull/3167/files#diff-1f94e63da3661a7cf4c493bbbb999c86R307), once you land this I'll rebase my changes on it.